bemuse/src/custom-folder/index.ts

```diff
@@ -283,6 +287,50 @@ async function updateFolders(
   }
 }
 
+async function removeFolders(
+  state: CustomFolderState,
+  handle: FileSystemDirectoryHandle,
+  io: CustomFolderScanIO
+): Promise<ScanIterationResult | undefined> {
+  const { log, setStatus } = io
+
+  if ((state.foldersToRemove?.length ?? 0) > 0) {
+    let remainingSongs
+    const foldersToRemove = [...state.foldersToRemove!]
+    const n = foldersToRemove.length
+    const removedPathSet = new Set<string>()
+    const deadline = Date.now() + 5000
+
+    for (const [i, folder] of foldersToRemove.entries()) {
+      removedPathSet.add(JSON.stringify(folder.path))
+
+      const pathStr = formatPath(folder.path)
+      const remaining = n - i
+      log(`Removing folder “${pathStr}” (${remaining} remaining)`)
+      const statusPrefix = `Folder “${pathStr}” (${remaining} remaining)`
+      setStatus(statusPrefix)
+
+      remainingSongs = state.songs?.filter((song) => {
```

Comment:
The test case only removes 1 folder, but this code may be buggy if you remove multiple songs, because the `remainingSongs` from previous iteration is discarded.

I recommend changing this

```suggestion
      remainingSongs = remainingSongs.filter((song) => {
```

And change above

```ts
    let remainingSongs = state.songs || []
```